qiskit/providers/ibmq/api/clients/auth.py

```diff
@@ -136,8 +149,50 @@ def user_hubs(self) -> List[Dict[str, str]]:
                     else:
                         hubs.append(entry)
 
+        # Check to see if the hub/group/project, specified by the `hub`, `group`,
+        # `project` fields of `credentials`, should be set as the default provider.
+        specified_hgp_index = self._get_default_hgp_index(credentials, hubs)
+        if isinstance(specified_hgp_index, int):
+            # Move the default hub/group/project to the front.
+            hubs[0], hubs[specified_hgp_index] = hubs[specified_hgp_index], hubs[0]
+
         return hubs
 
+    def _get_default_hgp_index(
+            self,
+            credentials: Credentials,
+            user_hubs: List[Dict[str, str]]
+    ) -> Optional[int]:
+        """Returns the index of the hub/group/project, specified by `credentials`, in `user_hubs`.
+
+        Args:
+            credentials: The `Credentials` containing the hub/group/project information of
+                the provider to set as the default.
+            user_hubs: A list of dictionaries with the hub, group, and project values keyed
+                by ``hub``, ``group``, and ``project``, respectively.
+
+        Returns:
+            The index of the hub/group/project, specified by `credentials`, in `user_hubs`,
+            else ``None``.
+        """
+        # The provider, as a dictionary, specified within `credentials`.
+        specified_hgp_entry = {'hub': credentials.hub,
+                               'group': credentials.group,
+                               'project': credentials.project}
```

Comment:
this h/g/p can now be in several format - as a string, dictionary, or embedded in a `Credential`. There's a existing `HubGroupProject` named tuple, maybe it's better to convert that into a class that has methods like `to_stored_format()`.